src/main/java/mateusz/michal/chat/Service/RegistrationService.java

```diff
@@ -28,8 +28,28 @@
     @Autowired
     SlugService slugService;
 
-    @Autowired
-    JsonFactory jsonFactory;
+    public ResponseEntity<IJsonResponse> getResponseForUserRegistration(UserDTO userDTO){
+        List<MyError> errors = validateRegistrationRequest(userDTO);
+        if (errors.size() == 0){
+            User user = createUser(userDTO);
+            userRepository.save(user);
+            return ResponseEntity.ok(
+                    JsonResponseFactory.createResponse(ResponseEnum.DATA,null,
+                            new SimpleDataResponse("user has been succesfully registered and saved in database"),
+                            null));
```

Comment:
Dlaczego tu takie duże wcięcia? Wszędzie masz 4 spacje a tutaj po 8 :o

Reply:
Tak to jest jak się łamie kod z w którym jest kilka zagnieżdżonych metod.
Najpierw metoda ok, następnie metoda createResponse.